drivers/pulse_counter/pulse_counter.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2017 UC Berkeley
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+
+/**
+ * @ingroup     drivers_pulse_counter
+ * @{
+ *
+ * @file
+ * @brief       Driver for the PULSE COUNTER.
+ *
+ * @author      Hyung-Sin Kim <hs.kim@cs.berkeley.edu>
+ *
+ * @}
+ */
+
+#include <string.h>
+
+#include "pulse_counter_params.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/* Accumulate pulse count */
+static void pulse_counter_trigger(void* arg) {
+    pulse_counter_t *dev = (pulse_counter_t*)arg;
+
+    /* Use atomic operations to avoid messing with IRQ flags */
+    __atomic_fetch_add(dev->pulse_count, 1, __ATOMIC_SEQ_CST);
+}
+
+/* Initialize pulse counter */
+int pulse_counter_init(pulse_counter_t *dev, const pulse_counter_params_t *params)
+{    
+    dev->p.name = params->name;
```

Comment:
`memcpy(&dev->p, params, sizeof(pulse_counter_params_t))`?

Reply:
or even better: there is no need to keep the params at all, they are not used after `gpio_init_int()` was called. So simply drop the `p` field from the struct I would say.

Reply:
sure. Thanks :)